See: https://bugs.gentoo.org/show_bug.cgi?id=578098

commit 240cd84b494e0ffee8ad261c43b927d246cf6be1
Author: Daniel Stenberg <daniel@haxx.se>
Date:   Wed Mar 23 10:04:48 2016 +0100

    openssl: fix ERR_remove_thread_state() for boringssl/libressl
    
    The removed arg is only done in OpenSSL
    
    Bug: https://twitter.com/xtraemeat/status/712564874098917376

diff --git a/lib/vtls/openssl.c b/lib/vtls/openssl.c
index cbf2d21..b7e4462 100644
--- a/lib/vtls/openssl.c
+++ b/lib/vtls/openssl.c
@@ -95,7 +95,9 @@
 
 #if (OPENSSL_VERSION_NUMBER >= 0x10000000L)
 #define HAVE_ERR_REMOVE_THREAD_STATE 1
-#if (OPENSSL_VERSION_NUMBER >= 0x10100004L)
+#if (OPENSSL_VERSION_NUMBER >= 0x10100004L) && \
+  !defined(LIBRESSL_VERSION_NUMBER) && \
+  !defined(OPENSSL_IS_BORINGSSL)
 /* OpenSSL 1.1.0-pre4 removed the argument! */
 #define HAVE_ERR_REMOVE_THREAD_STATE_NOARG 1
 #endif
